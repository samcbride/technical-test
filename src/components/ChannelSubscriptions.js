import React from "react";
import IndividualAccounts from "./IndividualAccounts";
import PropTypes from "prop-types";

const ChannelSubscriptions = ({ eligibility }) => (
  <div className="channel-subscriptions">
    {eligibility.map((eligible) => (
      <IndividualAccounts
        key={eligible.accountNumber}
        accountNumber={eligible.accountNumber}
        sports={eligible.channelSubscriptions.sports}
        kids={eligible.channelSubscriptions.kids}
        music={eligible.channelSubscriptions.music}
        news={eligible.channelSubscriptions.news}
        movies={eligible.channelSubscriptions.movies}
      />
    ))}
  </div>
);

export default ChannelSubscriptions;

ChannelSubscriptions.propTypes = {
  eligibility: PropTypes.arrayOf(
    PropTypes.shape({
      accountNumber: PropTypes.number.isRequired,
      channelSubscriptions: PropTypes.shape({
        sports: PropTypes.string,
        kids: PropTypes.string,
        music: PropTypes.string,
        news: PropTypes.string,
        movies: PropTypes.string,
      }).isOptional,
    })
  ).isRequired,
};
