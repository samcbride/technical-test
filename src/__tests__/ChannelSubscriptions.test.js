import React from "react";
import ChannelSubscriptions from "../components/ChannelSubscriptions";
import { render } from "@testing-library/react";

describe("ChannelSubscriptions", () => {
  const validProps = [
    {
      accountNumber: 1234,
      channelSubscriptions: {
        sports: "Sports",
        kids: "Kids",
        music: "Music",
        news: "News",
        movies: "Movies",
      },
    },
    {
      accountNumber: 5678,
      channelSubscriptions: {
        sports: "Sports",
        kids: "Kids",
        music: "Music",
        news: "News",
        movies: "Movies",
      },
    },
  ];
  it("renders correctly", () => {
    const { asFragment } = render(
      <ChannelSubscriptions eligibility={validProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of IndividualAccounts instances", () => {
    const { getAllByTestId } = render(
      <ChannelSubscriptions eligibility={validProps} />
    );
    expect(getAllByTestId("individual-channels")).toHaveLength(2);
  });
});
