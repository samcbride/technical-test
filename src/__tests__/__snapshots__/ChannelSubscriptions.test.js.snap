// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelSubscriptions renders correctly 1`] = `
<DocumentFragment>
  <div
    class="channel-subscriptions"
  >
    <div
      class="individual-channels"
      data-testid="individual-channels"
    >
      <div
        class="individual-channels__accountNumber"
      >
        1234
      </div>
      <div
        class="individual-channels__sports"
      >
        Sports
      </div>
      <div
        class="individual-channels__kids"
      >
        Kids
      </div>
      <div
        class="individual-channels__music"
      >
        Music
      </div>
      <div
        class="individual-channels__news"
      >
        News
      </div>
      <div
        class="individual-channels__movies"
      >
        Movies
      </div>
    </div>
    <div
      class="individual-channels"
      data-testid="individual-channels"
    >
      <div
        class="individual-channels__accountNumber"
      >
        1234
      </div>
      <div
        class="individual-channels__sports"
      >
        Sports
      </div>
      <div
        class="individual-channels__kids"
      >
        Kids
      </div>
      <div
        class="individual-channels__music"
      >
        Music
      </div>
      <div
        class="individual-channels__news"
      >
        News
      </div>
      <div
        class="individual-channels__movies"
      >
        Movies
      </div>
    </div>
  </div>
</DocumentFragment>
`;
